app/controllers/batch_controller.rb

```diff
@@ -62,6 +49,29 @@ def show
 
 private
 
+  def create_batch(create_params)
+    batch = Batch.create!(
+      webhook_uri: create_params.webhook_uri,
+      webhook_secret_token: create_params.webhook_secret_token
+    )
+
+    ActiveRecord::Base.transaction do
+      links = Link.fetch_all(create_params.uris)
+      checks = Check.fetch_all(links, within: create_params.checked_within)
+
+      batch_checks = checks.each_with_index.map do |check, i|
+        BatchCheck.new(batch_id: batch.id, check_id: check.id, order: i)
+      end
+
+      BatchCheck.import(batch_checks)
+    end
+
+    batch
+  rescue ActiveRecord::StatementInvalid, ActiveRecord::Rollback, ActiveRecord::RecordInvalid
```

Comment:
I assume any of these exceptions would cause the transaction to be automatically rolled back anyway.